### ISSUE TITLE
hicAdjustMatrix creates somehow malformed coolers

### ISSUE THREAD

**Issue by MasterprojectRK** [deeptools/HiCExplorer#468]
hicAdjustMatrix seems to output malformed coolers.

For the following example, hicexplorer 3.3.1 and a 1000kb HMEC matrix from Rao et al. were used, the latter downloaded from ftp://cooler.csail.mit.edu/coolers/hg19/Rao2014-HMEC-MboI-allreps-filtered.1000kb.cool

```
hicAdjustMatrix -m Rao2014-HMEC-MboI-allreps-filtered.1000kb.cool -o test.cool --action keep --chromosomes chr1
```
This works, but when calling cooler info on the output, I get:
```
cooler info test.cool
{
    "bin-size": 1000000,
    "bin-type": "fixed",
    "creation-date": "2019-12-05T19:51:25.764637",
    "format": Traceback (most recent call last):
  File "/X/miniconda3/envs/hicSymm/bin/cooler", line 11, in <module>
    sys.exit(cli())
  File "/X/miniconda3/envs/hicSymm/lib/python3.6/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/X/miniconda3/envs/hicSymm/lib/python3.6/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/X/miniconda3/envs/hicSymm/lib/python3.6/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/X/miniconda3/envs/hicSymm/lib/python3.6/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/X/miniconda3/envs/hicSymm/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/X/miniconda3/envs/hicSymm/lib/python3.6/site-packages/cooler/cli/_util.py", line 197, in decorated
    func(*args, **kwargs)
  File "/X/miniconda3/envs/hicSymm/lib/python3.6/site-packages/cooler/cli/info.py", line 62, in info
    json.dump(attrs_to_jsonable(dct), f, indent=4)
  File "/X/miniconda3/envs/hicSymm/lib/python3.6/json/__init__.py", line 179, in dump
    for chunk in iterable:
  File "/X/miniconda3/envs/hicSymm/lib/python3.6/json/encoder.py", line 430, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "/X/miniconda3/envs/hicSymm/lib/python3.6/json/encoder.py", line 404, in _iterencode_dict
    yield from chunks
  File "/X/miniconda3/envs/hicSymm/lib/python3.6/json/encoder.py", line 437, in _iterencode
    o = _default(o)
  File "/X/miniconda3/envs/hicSymm/lib/python3.6/json/encoder.py", line 180, in default
    o.__class__.__name__)
TypeError: Object of type 'bytes' is not JSON serializable
```
Called on the original matrix, however, I get:
```
cooler info Rao2014-HMEC-MboI-allreps-filtered.1000kb.cool
{
    "bin-size": 1000000,
    "bin-type": "fixed",
    "creation-date": "2016-02-26T22:41:50.736798",
    "format-url": "https://github.com/mirnylab/cooler",
    "format-version": 2,
    "genome-assembly": "hg19",
    "id": null,
    "library-version": "0.3.0",
    "nbins": 3114,
    "nchroms": 25,
    "nnz": 4085267
}
```

hicInfo does not show any problems, although I wonder why the genome assembly from the original file is missing:
```
hicInfo -m test.cool 
# Matrix information file. Created with HiCExplorer's hicInfo version 3.3.1
File:	test.cool
Date:	2019-12-05T19:51:25.764637
Genome assembly:	unknown
Size:	250
Bin_length:	1000000
Number of chromosomes:	1
Non-zero elements:	26,420
The following columns are available: ['chrom' 'start' 'end']


Generated by:	HiCMatrix-11
Cooler library version:	cooler-0.8.5
HiCMatrix url:	https://github.com/deeptools/HiCMatrix
```
hicInfo on the original file:
```
hicInfo -m Rao2014-HMEC-MboI-allreps-filtered.1000kb.cool
# Matrix information file. Created with HiCExplorer's hicInfo version 3.3.1
File:	Rao2014-HMEC-MboI-allreps-filtered.1000kb.cool
Date:	2016-02-26T22:41:50.736798
Genome assembly:	hg19
Size:	3,114
Bin_length:	1000000
Number of chromosomes:	25
Non-zero elements:	4,085,267
The following columns are available: ['chrom' 'start' 'end' 'weight']
```

**Comment by LeilyR** [comment]
Hi, You are right. I have also tested it and I got the same error. The reason , I suspect, comes from the fact that hicexplorer did not save the weight metadata. Basically it generated a cooler with balance = false. We will work on it.

**Comment by joachimwolff** [comment]
Should be fixed with release 0.8.7 of cooler.
